javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
would that `==` the `empty()` instance? I think we're checking for reference equality in case of non-match